app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -22,22 +22,34 @@
 import com.google.android.material.snackbar.Snackbar;
 import fr.free.nrw.commons.Media;
 import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.auth.SessionManager;
 import fr.free.nrw.commons.bookmarks.Bookmark;
 import fr.free.nrw.commons.bookmarks.pictures.BookmarkPicturesContentProvider;
 import fr.free.nrw.commons.bookmarks.pictures.BookmarkPicturesDao;
 import fr.free.nrw.commons.contributions.Contribution;
 import fr.free.nrw.commons.di.CommonsDaggerSupportFragment;
+import fr.free.nrw.commons.mwapi.OkHttpJsonApiClient;
 import fr.free.nrw.commons.utils.DownloadUtils;
 import fr.free.nrw.commons.utils.ImageUtils;
 import fr.free.nrw.commons.utils.NetworkUtils;
 import fr.free.nrw.commons.utils.ViewUtil;
+import io.reactivex.disposables.CompositeDisposable;
+import java.util.Objects;
 import javax.inject.Inject;
 import timber.log.Timber;
 
 public class MediaDetailPagerFragment extends CommonsDaggerSupportFragment implements ViewPager.OnPageChangeListener {
 
     @Inject BookmarkPicturesDao bookmarkDao;
 
+    @Inject
+    OkHttpJsonApiClient okHttpJsonApiClient;
+
+    @Inject
```

Comment:
private?

Reply:
We can't make them private if I do I get this error

```private SessionManager sessionManager;
                           ^\apps-android-commons\app\src\main\java\fr\free\nrw\commons\di\CommonsApplicationComponent.java:44: error: [ComponentProcessor:MiscError] dagger.internal.codegen.ComponentProcessor was unable to process this interface because not all of its dependencies could be resolved. Check for compilation errors or a circular dependency with generated code.
public interface CommonsApplicationComponent extends AndroidInjector<ApplicationlessInjection> {
warning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIXwarning: unknown enum constant Scope.LIBRARY_GROUP_PREFIX[WARN] Incremental annotation processing requested, but support is disabled because the following processors are not incremental: butterknife.compiler.ButterKnifeProcessor (NON_INCREMENTAL).

> Task :app:kaptProdDebugKotlin FAILED```

Reply:
Interesting, thanks! How about `protected`?